### DISCUSSION TITLE
position non linearity

### DISCUSSION THREAD

**Discussion by luzik** [nagyrobi/home-assistant-custom-components-cover-rf-time-based#46] [Ideas]
My roller shutter moves from full up to down in 20second, but there is extra 3sec move down that locks shutter. Those 50% is far away from 50% shutter clearance.

Tasmota for example allows to configure calibration points
https://tasmota.github.io/docs/Blinds-and-Shutters/#calibration

**Comment by Jarnsen** [comment]
I think one approach to solving the problem would be to work with 2 additional times.
In full it could look like this:

 0% shutter is open
100% shutter is closed but slats still let light through
150% roller shutters completely closed, no more light comes in.

The times are measured from 0 to 100 and from 100 to 150 and vice versa, of course, from 150 to 100 and from 100 and to 0. The values ​​from 100 to 150 and from 150 to 100 are then the lag and the lead for calculating the exact position. And 50% should be in the middle of the window.

Is it possible to implement this?

Here is an example from FHEM

[KLICK](https://github.com/mhop/fhem-mirror/blob/master/fhem/FHEM/98_Dooya.pm)
